src/compiler/emitter.ts

```diff
@@ -4635,6 +4635,9 @@ namespace ts {
                         if (reservedInNestedScopes) {
                             reserveNameInNestedScopes(name);
                         }
+                        else {

```

Comment:
I don't think this is the correct fix. We want to reuse temporary variables in nested scopes whenever possible, otherwise we will quickly exceed the set of small (two-character) temporary variables we normally allocate and end up with larger and larger temp variable names.